sleuth/sleuth/enrich_tls.py

```diff
@@ -169,76 +107,87 @@ def tls_seclevel(scs, client_key_length, certs):
-                    else:
-                        tmp_seclevel = 2
-                elif sig_alg == "sha256WithRSAEncryption":
-                    if sig_key_size < 1024:
-                        tmp_seclevel = 1
-                    elif sig_key_size < 2048:
-                        tmp_seclevel = 2
-                    else:
-                        tmp_seclevel = 3
-                elif sig_alg == "ecdsa-with-SHA256":
-                    if sig_key_size < 832:
-                        tmp_seclevel = 1
-                    else:
-                        tmp_seclevel = 3
-                elif sig_alg == "sha384WithRSAEncryption" or sig_alg == "sha512WithRSAEncryption":
-                    if sig_key_size < 1024:
-                        tmp_seclevel = 1
-                    elif sig_key_size < 2048:
-                        tmp_seclevel = 2
-                    elif sig_key_size < 3072:
-                        tmp_seclevel = 3
-                    else:
-                        tmp_seclevel = 4
+                tmp_seclevel = certs_seclevel
+
+                if sig_alg and sig_alg in sig_policy:
+                    if sig_key_size and "sig_key_size" in sig_policy[sig_alg]:
+                        for each in sig_policy[sig_alg]["sig_key_size"]:
+                            if sig_key_size > int(each):
+                                tmp_seclevel = sig_policy[sig_alg]["sig_key_size"][each]
+                                break
+                    elif "seclevel" in sig_policy[sig_alg]:
+                        tmp_seclevel = sig_policy[sig_alg]["seclevel"]
                 else:
-                    tmp_seclevel = 0
-                if tmp_seclevel < certs_seclevel:
+                    tmp_seclevel = UNKNOWN
+
+                if tmp_seclevel and tmp_seclevel < certs_seclevel:
                     certs_seclevel = tmp_seclevel
         else:
-            certs_seclevel = 0
+            certs_seclevel = UNKNOWN
+
+        # Other seclevel items can be assessed in one step due to the lack of nested
+        # seclevel elements; we create an inventory for the flow here
+        #
+        seclevel_inventory = {
+            "kex": kex_seclevel,
+            "certs": certs_seclevel,
+            "sig": policy_rules["sig"][params['sig']] if params['sig'] in policy_rules["sig"] else UNKNOWN,
+            "enc": policy_rules["enc"][params['enc']] if params['enc'] in policy_rules["enc"] else UNKNOWN,
+            "auth": policy_rules["auth"][params['auth']] if params['auth'] in policy_rules["auth"] else UNKNOWN,
+            "hash": policy_rules["hash"][params['hash']] if params['hash'] in policy_rules["hash"] else UNKNOWN
+        }
+
+        # fetch min seclevel element based on whether or not 'unknown' elements should be reported.
+        # default here is 'report'
+        #
+        if unknowns == "ignore":
+            seclevel_floor = min({ k: v for k, v in seclevel_inventory.items() if v }.values())
```

Comment:
You should be able to simplify this to min([ v for k, v in seclevel_inventory.iteritems()])